### ISSUE TITLE
Request changes on pull request results in server error 500

### ISSUE THREAD

**Issue by ChriFo** [go-gitea/gitea#7205]
- Gitea version: 1.8.2
- Operating system: official docker setup
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant

## Description

Submitting a long comment to a code review with the button "request changes" leads into a 500 page. The following line in the file gitea.log appears in that moment:

` 2019/06/14 13:20:11 [.../repo/pull_review.go:163 SubmitReview()] [E] CreateReview: pq: value too long for type character varying(255)`

I was able to fix the issue by manual changing the column type of `content` in the table `public.review` to just varchar (without limit).

The gitea setup is 6 month old and is continuously updated when a new version is arriving.

Maybe an other migration file is needed to have these changes for other "long-time" users as well.

**Comment by jeansergegagnon** [comment]
Not sure if this is same issue, but any pull request we have created shows a 500 error when trying to see the pull request once it is merged.

We are using Gitea Version: 1.9.0+dev-376-ge07ff2f89 and the log shows:
```
2019/06/25 11:01:14 routers/repo/pull.go:302:PrepareMergedViewPullInfo() [E] GetCompareInfo: exit status 128 - fatal: ambiguous argument 'b568d59d88af7ea7393d6ae2c9da09794a3c136d...refs/pull/4/head': unknown revision or path not in the working tree.
        Use '--' to separate paths from revisions, like this:
        'git <command> [<revision>...] -- [<file>...]'

```

**Comment by Ahaus314** [comment]
I have the same issue with version 1.9.4.
The interface say everything should be fine and when I try to execute the merge, I got an error 500 and this is the error message:
2019/10/15 11:16:20 routers/repo/pull.go:629:MergePullRequest() [E] Merge: git merge --no-ff --no-commit [<…>/ocal-repo/merge-663207900.git]: exit status 128 - From <REPO>
	 * [new branch]            <BRANCH> -> head_repo/<BRANCH>	error: Cannot update sparse checkout: the following entries are not up-to-date:
<LIST OF FILES>

**Comment by guillep2k** [comment]
Maybe this can help?
https://stackoverflow.com/a/56978142/225979

**Comment by zeripath** [comment]
@ahaus314 this is not the same issue. Your error is on merging not on adding a request changes comment.

You likely need to try a different merge strategy but I recommend you open a different issue.

**Comment by sbradl** [comment]
> I have the same issue with version 1.9.4.
> The interface say everything should be fine and when I try to execute the merge, I got an error 500 and this is the error message:
> 2019/10/15 11:16:20 routers/repo/pull.go:629:MergePullRequest() [E] Merge: git merge --no-ff --no-commit [<…>/ocal-repo/merge-663207900.git]: exit status 128 - From
> * [new branch] -> head_repo/ error: Cannot update sparse checkout: the following entries are not up-to-date:

Got the same error and opened a new issue https://github.com/go-gitea/gitea/issues/8744

**Comment by lunny** [comment]
I think this has been resolved and please feel free to reopen it.
